latest/index.bs

```diff
@@ -376,16 +377,20 @@ custom attributes of the plate group under the `plate` key.
   <dd>A list of JSON objects defining the rows of the plate. Each row object
       defines the properties of the row at the index of the object in the
       list. If not empty, it MUST contain a `name` key specifying the row
-      name.</dd>
+      name. Each row in the physical plate MUST be defined, even if no wells
+      in the row are defined.</dd>
   <dt><strong>version</strong></dt>
   <dd>A string defining the version of the specification.</dd>
   <dt><strong>wells</strong></dt>
   <dd>A list of JSON objects defining the wells of the plate. Each well object 
-      MUST contain a `path` key identifying the path to the well subgroup.</dd>
+      MUST contain a `path` key identifying the path to the well subgroup.
+      Each well object MUST contain both a `row_index` key identifying the index into
```

Comment:
Regarding https://github.com/ome/ngff/pull/24#issuecomment-801297872, are there cases where it is not possible to recompute these indexes based on the knowledge of the individual wells `path` as well as the `rows` and `names` dictionaries? If recomputing is always possible (but at the cost of the consumer), my primary consideration is whether the recommendation for these  new fields should be `SHOULD` rather than `MUST`.

For real-world examples, I can definitely see how `row_index/column_index` makes sense in terms of optimizing some of the queries. In addition to testing this with sparse plates, it will be useful to also generate representative plate with many wells (384 at least) to check there is no performance impact with the extra JSON metadata.

Reply:
In order for these indexes to be forward or reverse computable, `path` would need to be much more explicitly defined than it is now:

> A list of JSON objects defining the fields of views for a given well. Each object MUST contain a path key identifying the path to the field of view. If multiple acquisitions were performed in the plate, it SHOULD contain an acquisition key identifying the id of the acquisition which must match one of acquisition JSON objects defined in the plate metadata.

Furthermore, the `wells` array would need have be null or similar padding in order for those indexes to make sense.

Neither of these things are ideal obviously. I don't think there's a way to not have these things be `MUST` if we want to guarantee that lookups can happen based on physical plate characteristics.